cpp/dolfinx/mesh/graphbuild.cpp

```diff
@@ -147,50 +150,44 @@ compute_nonlocal_dual_graph(
   const int num_facets_rcvd
       = recvd_buffer.array().size() / (max_num_vertices_per_facet + 1);
 
-  // Build array from received facet to source rank
-  const std::vector<std::int32_t>& recvd_disp = recvd_buffer.offsets();
-  std::vector<int> proc(num_facets_rcvd);
-  for (int p = 0; p < num_ranks; ++p)
-  {
-    for (int f = recvd_disp[p] / (max_num_vertices_per_facet + 1);
-         f < recvd_disp[p + 1] / (max_num_vertices_per_facet + 1); ++f)
-    {
-      proc[f] = p;
-    }
-  }
+  // Create a return buffer with one entry per received facet
```

Comment:
Do you mean receive buffer?

Reply:
It's a reply to the received data. Maybe `reply_buffer` is better. The sequence is something like send -> recv, process data, then reverse the communication, sending back a reply to the originator.